packages/ceramic-core/src/__tests__/ceramic-anchor.test.ts

```diff
@@ -0,0 +1,347 @@
+
+    const doctype = await ceramic1.createDocument(DOCTYPE_TILE, { content: { x: 1 } }, { applyOnly: false })
+
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: true })
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: false })
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    await ceramic1.context.anchorService.anchor()
+
+    const updatePromise = new Promise(resolve => {
+      doctype.on('change', () => {
+        resolve()
+      })
+    })
+
+    await updatePromise
+
+    expect(doctype.content).toEqual({ x: 1 })
+    expect(doctype.state.log.length).toEqual(2)
+
+    const doctype2 = await ceramic2.loadDocument(doctype.id)
+    expect(doctype.content).toEqual(doctype2.content)
+    expect(doctype.state.log.length).toEqual(doctype2.state.log.length)
+
+    await ceramic1.close()
+    await ceramic2.close()
+  })
+
+  it('test last one anchored', async () => {
+    await ipfs2.swarm.connect(multaddr1)
+
+    const ceramic1 = await createCeramic(ipfs1, true, topic)
+    const ceramic2 = await createCeramic(ipfs2, false, topic)
+    const owner = ceramic1.context.did.id
+
+    const doctype = await ceramic1.createDocument(DOCTYPE_TILE, { content: { x: 1 } }, { applyOnly: true })
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: true })
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: false })
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    await ceramic1.context.anchorService.anchor()
+
+    const updatePromise = new Promise(resolve => {
+      doctype.on('change', () => {
+        resolve()
+      })
+    })
+
+    await updatePromise
+
+    expect(doctype.content).toEqual({ x: 3 })
+    expect(doctype.state.log.length).toEqual(4)
+
+    const doctype2 = await ceramic2.loadDocument(doctype.id)
+    expect(doctype.content).toEqual(doctype2.content)
+    expect(doctype.state.log.length).toEqual(doctype2.state.log.length)
+
+    await ceramic1.close()
+    await ceramic2.close()
+  })
+
+  it.skip('in the middle anchored', async () => {
+    await ipfs2.swarm.connect(multaddr1)
+
+    const ceramic1 = await createCeramic(ipfs1, true, topic)
+    const ceramic2 = await createCeramic(ipfs2, false, topic)
+    const owner = ceramic1.context.did.id
+
+    const doctype = await ceramic1.createDocument(DOCTYPE_TILE, { content: { x: 1 } }, { applyOnly: true })
+
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: false })
+    await doctype.change({ content: { x: doctype.content.x + 1 }, metadata: { owners: [owner] } }, { applyOnly: true })
+
+    let updatePromise = new Promise(resolve => {
+      doctype.on('change', () => {
```

Comment:
Maybe it's not an issue in this case, but considering the `doctype` is reused and can emit the `change` event multiple times, using `.once()` rather than `.on()` could be better suited for this purpose.

Reply:
Yep, makes sense!